### PR TITLE
Update copyright header template to be entity agnostic, update existing headers

### LINKED ISSUE #902: Update copyright headers
**Describe the problem/challenge you have**
Currently the copyright headers for all the tools say:
```
// Copyright 2023 VMware, Inc.
// SPDX-License-Identifier: Apache-2.0
```

**Describe the solution you'd like**
Now that Carvel is a CNCF project, it should be something like
```
// Copyright 2023 The Carvel Authors.
// SPDX-License-Identifier: Apache-2.0
```

---
Vote on this request

This is an invitation to the community to vote on issues, to help us prioritize our backlog. Use the "smiley face" up to the right of this comment to vote.

👍 "I would like to see this addressed as soon as possible"
👎 "There are other more important things to focus on right now"

We are also happy to receive and review Pull Requests if you want to help working on this issue.

### PR DESCRIPTION
**Pull request by everettraven**
#### What this PR does / why we need it:
- Updates the required code copyright header template to be entity agnostic
- Updates all the existing headers to use the new template and updates copyright year to 2024
#### Which issue(s) this PR fixes:

Fixes #902 

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

##### Review Checklist:

- [ ] Follows the [developer guidelines](https://carvel.dev/shared/docs/latest/development_guidelines/)
- [ ] Relevant tests are added or updated
- [ ] Relevant docs in this repo added or updated
- [ ] Relevant carvel.dev docs added or updated in a separate PR and there's
  a link to that PR
- [ ] Code is at least as readable and maintainable as it was before this
  change

#### Additional documentation e.g., Proposal, usage docs, etc.:

```docs

```

### CODE REVIEW

File: cmd/kapp/kapp.go [hunk 1]
```diff
@@ -1,4 +1,4 @@
-// Copyright 2020 VMware, Inc.
+// Copyright 2024 The Carvel Authors.
```
Comment (everettraven) [inline cmd/kapp/kapp.go:1]: I updated all the copyright years to 2024 as I figured that would be required. If folks feel that is unnecessary or is not best practice/allowed I'll revert the copyright years and only change from `VMWare, Inc` --> `The Carvel Authors`

Comment (kumaritanushree) [reply cmd/kapp/kapp.go:1]: May I know why do you feel it will be required to change year? As per my understanding we add the year as evidence of when the file has been written.

Comment (everettraven) [reply cmd/kapp/kapp.go:1]: I've seen this done in quite a few open source projects in the past, but I am by no means an expert in copyright which is why if there is someone more familiar with this that says this isn't necessary I am happy to revert it.

As far as I am aware though, copyright should be updated whenever changes are "published" to a file to make sure the changes are appropriately captured in the copyright. This is totally subjective, but since this is an open source project a commit being merged is, I believe, the same as that change being "published". I think in general it is easier to just do a wider sweep and update all the copyrights to the most recent year. From a bit of surface level research there are various valid formats for copyright updates. A brief set of what I found is valid:
- `Copyright 2020, 2022, 2024 ...`
- `Copyright 2020 - 2024 ...`
- `Copyright 2024 ...`

Just to re-iterate, I am not an expert and going based on prior experience and a very rudimentary understanding of copyright when it comes to open source projects. I am happy to revert the changes to the year if someone with more knowledge/experience says that the changes are unnecessary.

Comment (praveenrewar) [reply cmd/kapp/kapp.go:1]: I think it makes sense to switch to the 2024. @microwavables, @joaopapereira Do you have any thoughts/concerns with it?

File: pkg/kapp/config/default_test.go [hunk 2]
```diff
@@ -105,14 +106,14 @@ spec:
   0,  3 +   annotations:
   0,  4 +     kapp.k14s.io/versioned: ""
   0,  5 +   name: test-configmap-ver-1
-  0,  6 + 
+  0,  6 +
```
Comment (praveenrewar) [inline pkg/kapp/config/default_test.go]: I think this is causing the tests to fail. Was this change intended?

File: cmd/kapp/kapp.go (hunk 1 above)
Comment (joaopapereira) [reply cmd/kapp/kapp.go:1]: I am ok with the move to 2024

File: pkg/kapp/config/default_test.go (hunk 2 above)
Comment (everettraven) [reply pkg/kapp/config/default_test.go]: Ah, not sure why that changed - probably an autoformatting thing with the editor I was using at the time :( - Will try to get that fixed in the next couple of days.

Comment (everettraven) [reply pkg/kapp/config/default_test.go]: This should be fixed with the latest push - thanks for catching! 

Lesson learned to not use my editor to do a find + replace and instead use `sed` or equivalent 🤦‍♂️

**Review by praveenrewar — APPROVED** [review]
LGTM! Thank you for the change 🚀
